### PR TITLE
cob4 teleop working

### PR DESCRIPTION
**Pull request by ipa-fmw-ms**
New:
init/recover/move with script server
all components (except gripper)
TODO:
Stop homing if deadman released
init recover for all components
...

### CODE REVIEW

File: cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp [hunk 1]
```diff
@@ -191,92 +241,99 @@ class cob_teleop_cob4_impl
...
 
     /* protected region user additional functions on begin */
-    //Todo: overload parameter and function, so 2Inputs for pair of buttons, one for axis
-  float run_factor(cob_teleop_cob4_data &data) //dont need a function for this, will be removed
+    void init_recover(std::string component)
   {
-  float result;
-  result = 1-data.in_joy.axes.at(9);//value between 1 and 2
-  return result;
+    sss.component_name=(component);
```
Comment (floweisshardt) [inline cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp]: remove brackets

File: cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp [hunk 2]
```diff
@@ -92,46 +105,74 @@ class cob_teleop_cob4_data
...
-      right.points.resize(1);
-      right.points.at(0).velocities.resize(7);
+      mode=0;
+      jvalue.unit="rad/sec";
+      left.velocities.resize(7);
+      right.velocities.resize(7);
+      int i;
+      for (i=0; i<7; i++)
+      {
+      //jvalue.joint_uri=("arm_left_%d_joint",(i+1)); //produces Ascii crap like:  
```
Comment (floweisshardt) [inline cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp]: remove

File: cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp [hunk 3]
```diff
@@ -92,46 +105,74 @@ class cob_teleop_cob4_data
...
+      left.velocities.at(i)=jvalue;
+      //jvalue.joint_uri=("arm_right_%d_joint",(i+1));
+      right.velocities.at(i)=jvalue;
+      }
+      left.velocities[0].joint_uri="arm_left_1_joint";
```
Comment (floweisshardt) [inline cob_teleop_cob4/common/src/cob_teleop_cob4_common.cpp:160]: BRIDE 3.1 support list parameters using XMLRPC values

**Comment by NadiaHG** [comment]
Visual Review: OK

**Pull request merged**